src/textual/_wrap.py

```diff
@@ -0,0 +1,147 @@
+from __future__ import annotations
+
+import re
+from typing import Iterable
+
+from rich.cells import get_character_cell_size
+
+from ._cells import cell_len
+from ._loop import loop_last
+from .expand_tabs import get_tab_widths
+
+re_chunk = re.compile(r"\S+\s*|\s+")
+
+
+def chunks(text: str) -> Iterable[tuple[int, int, str]]:
+    """Yields each "chunk" from the text as a tuple containing (start_index, end_index, chunk_content).
+    A "chunk" in this context refers to a word and any whitespace around it.
+
+    Args:
+        text: The text to split into chunks.
+
+    Returns:
+        Yields tuples containing the start, end and content for each chunk.
+    """
+    end = 0
+    while (chunk_match := re_chunk.match(text, end)) is not None:
+        start, end = chunk_match.span()
+        chunk = chunk_match.group(0)
+        yield start, end, chunk
+
+
+def compute_wrap_offsets(
```

Comment:
This looks expensive. Could we cache this?

Not suggesting a change. Just thinking out loud.